build.gradle

```diff
@@ -10,40 +10,49 @@
 
 buildscript {
     repositories {
-        mavenLocal()
-        //mavenCentral()
-        maven { url "https://repo1.maven.org/maven2" }
-        //maven { url "https://repo.grails.org/grails/core" }
-        maven { url "https://repo.grails.org/grails/plugins/" }
-        maven { url "https://repo.grails.org/grails/plugins-releases/" }
-        maven { url "https://repo.grails.org/grails/core/" }
+        mavenCentral()
     }
     dependencies {
-        classpath "org.grails:grails-gradle-plugin:$grailsVersion"
-        classpath "org.grails.plugins:hibernate5:${gormVersion-".RELEASE"}"
-        classpath "com.bertramlabs.plugins:asset-pipeline-gradle:2.15.1"
-        classpath "gradle.plugin.com.gorylenko.gradle-git-properties:gradle-git-properties:2.2.0"
-        classpath 'org.grails.plugins:database-migration:3.1.0.RC1'
-        classpath "gradle.plugin.com.virgo47.gradle:ClasspathJar:1.0.0"
-
+        classpath "com.bertramlabs.plugins:asset-pipeline-gradle:${assetPipelineVersion}"
+        classpath "io.swagger.core.v3:swagger-gradle-plugin:${swaggerCoreVersion}"
+        classpath "org.grails:grails-gradle-plugin:${grailsPluginVersion}"
     }
 }
 
-//include gradle-node plugin
 plugins {
-    id "com.moowork.node" version "0.12"
+    id "eclipse"
+    id "idea"
+    id "java"
+    id "war"
+    id "com.gorylenko.gradle-git-properties" version "2.2.0"
+    id "com.moowork.node" version "1.2.0"
     id "com.virgo47.ClasspathJar" version "1.0.0"
+    // versions 2.19.x seem to require Java 11 or 17
+    id "org.hidetake.swagger.generator" version "2.18.2"
+}
+
+allprojects {
+    repositories {
+        mavenCentral()
+        maven { url "https://plugins.gradle.org/m2/" }
+        maven { url "https://repo.grails.org/grails/core" }
+        // needed only for barcode4j
+        maven { url "https://repo.grails.org/grails/plugins" }
+    }
 }
 
-//grails.project.class.dir = "target/classes"
-//grails.project.docs.output.dir = "web-app/docs"
-//grails.project.test.class.dir = "target/test-classes"
-//grails.project.test.reports.dir = "target/test-reports"
-//
-//grails.plugin.location.liquibase = "liquibase/"
+apply plugin: "com.bertramlabs.asset-pipeline"
+apply plugin: "io.swagger.core.v3.swagger-gradle-plugin"
+apply plugin: "org.grails.grails-gsp"
+apply plugin: "org.grails.grails-web"
 
-version "0.9.0-SNAPSHOT"
+version "0.9.0-SNAPSHOT"  // FIXME get this from git
```

Comment:
I don't think that's true. this is a pre-release version. version tags don't exist until after we release.

Reply:
I know in the python world there are plugins that if you're a descendant, of, say, 1.0.1, will generate build numbers like 1.0.2dev1 -- that's what I had in mind here

Reply:
but it's a minor nit and I can remove the comment if you prefer